### PR TITLE
[INTEL MKL] Enabling simple heuristic based tuning for innerproduct primitive

### PR DESCRIPTION
**Pull request by Srini511**
This PR enables a simple heuristic based approach to determine whether inner product primitive should run single threaded or not. If the matmul fits in L2 we have observed that single threaded operation gives better performance than multi-threaded.

### CODE REVIEW

File: tensorflow/core/kernels/mkl/mkl_matmul_ops_common.h [hunk 1]
```diff
@@ -32,18 +32,19 @@ using mkldnn::prop_kind;
...
-  // total size is < few L1's. So we will do this simple calculation
+  // total size is < L2. So we will do this simple calculation
   // to determine if the matrix multiplication should be run on a single thread.
-  constexpr int kHeuristicMultiplier = 8;
-  return ((sizeof(float) * (m * n + k * (m + n))) <
-          L1_SIZE * kHeuristicMultiplier);
+  ptrdiff_t l2_size = cache_sizes.m_l2;
+  constexpr int kHeuristicMultiplier = 1;
+  int mul_size = bytes * (m * n + k * (m + n));
+  int l2_heur = l2_size * kHeuristicMultiplier;
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl/mkl_matmul_ops_common.h:46]: These can be `const int`.

File: tensorflow/core/util/mkl_threadpool.h [hunk 2]
```diff
@@ -63,12 +63,12 @@ inline void balance211(T n, U team, U tid, T* n_start, T* n_end) {
 struct MklDnnThreadPool : public threadpool_iface {
   MklDnnThreadPool() = default;
 
-  MklDnnThreadPool(OpKernelContext* ctx)
-      : eigen_interface_(ctx->device()
-                             ->tensorflow_cpu_worker_threads()
-                             ->workers->AsEigenThreadPool()) {}
+  MklDnnThreadPool(OpKernelContext* ctx, int num_threads = -1) {
+    eigen_interface_ = ctx->device()->tensorflow_cpu_worker_threads()->workers->AsEigenThreadPool();
+    num_threads_ = (num_threads == -1) ? eigen_interface_->NumThreads(): num_threads;
```
Comment (penpornk) [inline tensorflow/core/util/mkl_threadpool.h:68]: Please reformat the source.

**Review by penpornk — CHANGES_REQUESTED** [review]

File: tensorflow/core/util/mkl_threadpool.h (hunk 2 above)
Comment (Srini511) [reply tensorflow/core/util/mkl_threadpool.h]: @penpornk Addressed your comments. Please have a look.

**Review by penpornk — APPROVED** [review]
Thank you very much for the quick changes!

**Pull request merged**